### ISSUE TITLE
Automatic dune file formatting on save doesn't respect (lang dune _version_)

### ISSUE THREAD

**Issue by ulugbekna** [ocamllabs/vscode-ocaml-platform#526]
_Problem:_ 

Current formatting of dune files is done by using a plumbing command `dune format-dune-file`, which doesn't care about `(lang dune ...)` stanza, and latest `dune` (>= v2.8) has different dune file formatting than previous versions. This creates a problem: 

If a user has `dune v2.8`, but there is `(lang dune 2.0)` set in `dune-project`, they end up with conflicting formatting, i.e., on auto-save their dune file is formatted with `dune 2.8` but `dune build @fmt` formats using `dune v2.0`. (If I understand correctly, upgrading `lang dune` to `2.8` will require dep `dune >= 2.8`)

_Solution:_ 

In short term, we can parse the first line of `dune-project` to respect dune version in formatting. 
In long term, when we get dune rpc, we could use that.

**Comment by Khady** [comment]
Sounds related to https://github.com/ocaml/dune/issues/3516

**Comment by tmattio** [comment]
As you said @ulugbekna, we're just calling `dune format-dune-file`, so I think the problem would be better addressed in Dune directly.

**Comment by tmattio** [comment]
We have a tag to track things to be addressed upstreamed, but I don't think we'll want to actively work on this to address the problem in dune, so I'm going to close this and the feature can be tracked in ocaml/dune#3516.
